src/dbnode/storage/bootstrap/bootstrapper/commitlog/source_instrumentation.go

```diff
@@ -0,0 +1,150 @@
+// Copyright (c) 2021 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package commitlog
+
+import (
+	"time"
+
+	"github.com/opentracing/opentracing-go"
+	opentracinglog "github.com/opentracing/opentracing-go/log"
+	"github.com/uber-go/tally"
+	"go.uber.org/zap"
+
+	"github.com/m3db/m3/src/dbnode/storage/profiler"
+	"github.com/m3db/m3/src/dbnode/tracepoint"
+	"github.com/m3db/m3/src/x/clock"
+	"github.com/m3db/m3/src/x/context"
+)
+
+type instrumentationContext struct {
+	opts                       Options
+	nowFn                      clock.NowFn
+	log                        *zap.Logger
+	start                      time.Time
+	span                       opentracing.Span
+	bootstrapSnapshotsDuration tally.Timer
+	bootstrapCommitLogDuration tally.Timer
+	pOpts                      profiler.Options
+	pCtx                       profiler.ProfileContext
+}
```

Comment:
I understand that PR got a bit bigger because of this but the reason I've included it in this PR is because instrumentation metrics plays well with additional profiling, meaning profiling is performed inside instrumentation methods. We would lose this concept if instrumentation would be extracted into separate PR.